jobbergate-api/jobbergate_api/apps/applications/routers.py

```diff
@@ -22,42 +23,55 @@
 
 
 @router.post(
-    "/applications/", status_code=status.HTTP_201_CREATED, description="Endpoint for application creation",
+    "/applications/",
+    status_code=status.HTTP_201_CREATED,
+    response_model=ApplicationResponse,
+    description="Endpoint for application creation",
 )
 async def applications_create(
-    application_name: str = Form(...),
-    application_identifier: Optional[str] = Form(None),
-    application_description: str = Form(""),
-    application_config: str = Form(...),
-    application_file: str = Form(...),
+    application: ApplicationCreateRequest,
     token_payload: TokenPayload = Depends(guard.lockdown("jobbergate:applications:create")),
-    upload_file: UploadFile = File(...),
 ):
     """
     Create new applications using an authenticated user token.
     """
     armada_claims = ArmadaClaims.from_token_payload(token_payload)
-    application = ApplicationRequest(
-        application_name=application_name,
-        application_identifier=application_identifier,
-        application_description=application_description,
-        application_file=application_file,
-        application_config=application_config,
-        application_owner_email=armada_claims.user_email,
-    )
+    application.application_owner_email = armada_claims.user_email
 
     async with database.transaction():
         try:
-            query = applications_table.insert()
+            insert_query = applications_table.insert()
             values = application.dict()
-            application_created_id = await database.execute(query=query, values=values)
+            inserted_id = await database.execute(query=insert_query, values=values)
 
         except INTEGRITY_CHECK_EXCEPTIONS as e:
             raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
 
-    s3man.put(upload_file, app_id=application_created_id)
+        # Now fetch the newly inserted row. This is necessary to reflect defaults and db modified columns
+        query = applications_table.select().where(applications_table.c.id == inserted_id)
+        raw_application = await database.fetch_one(query)
+        response_application = ApplicationResponse.parse_obj(raw_application)
+
+    return response_application
+
 
-    return Application(id=application_created_id, **application.dict())
+@router.post(
+    "/applications/{application_id}/upload",
+    status_code=status.HTTP_201_CREATED,
+    description=(
+        "Endpoint for uploading application files. "
+        "The file should be a gzipped tar-file (e.g. `jobbergate.tar.gz`)."
+    ),
+    dependencies=[Depends(guard.lockdown("jobbergate:applications:upload"))],
+)
+async def applications_upload(
+    application_id: int = Query(..., description="id of the application for which to upload a file"),
+    upload_file: UploadFile = File(..., description="The gzipped application tar-file to be uploaded"),
+):
```

Comment:
do we have any size restrictions here? I ask because the infrastructure itself does have. ALBs support 100 MB maximum as payload.

Reply:
Added limit on upload file size and unit tests to make sure it works.
Change in latest commit.

Reply:
good job!